resources/assets/lib/wiki-search-controller.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ *    Copyright (c) ppy Pty Ltd <contact@ppy.sh>.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { route } from 'laroute';
+import { debounce } from 'lodash';
+import { action, computed, observable } from 'mobx';
+
+interface SuggestionJSON {
+  highlight: string;
+  path: string;
+  title: string;
+}
+
+export class WikiSearchController {
+  @observable selectedIndex = -1;
+  @observable shouldShowSuggestions = false;
+  @observable suggestions = observable.array<SuggestionJSON>([]);
```

Comment:
I just noticed, the `@observable` decorator isn't needed here, is it?

Reply:
Technically, `@observable` is implicitly wrapping the array and `observable.array` is just there to infer the correct type so we can reassign the variable later without typescript errors